go-controller/pkg/util/multinetwork.go

```diff
@@ -0,0 +1,262 @@
+		klog.V(5).Infof("New netconf topology type is different, expect %s",
+			layer3NetConfInfo.GetTopologyType())
+		return false
+	}
+	if layer3NetConfInfo.NetCidr != newLayer3NetConfInfo.NetCidr {
+		klog.V(5).Infof("New netconf NetCidr %v has changed, expect %v",
+			newLayer3NetConfInfo.NetCidr, layer3NetConfInfo.NetCidr)
+		return false
+	}
+	if layer3NetConfInfo.MTU != newLayer3NetConfInfo.MTU {
+		klog.V(5).Infof("New netconf MTU %v has changed, expect %v",
+			newLayer3NetConfInfo.MTU, layer3NetConfInfo.MTU)
+		return false
+	}
+	return true
+}
+
+func (layer3NetConfInfo *Layer3NetConfInfo) Verify() error {
+	clusterSubnets, err := config.ParseClusterSubnetEntries(layer3NetConfInfo.NetCidr)
+	if err != nil {
+		return fmt.Errorf("cluster subnet %s is invalid: %v", layer3NetConfInfo.NetCidr, err)
+	}
+	layer3NetConfInfo.ClusterSubnets = clusterSubnets
+	return nil
+}
+
+func (layer3NetConfInfo *Layer3NetConfInfo) GetTopologyType() string {
+	return types.Layer3AttachDefTopoType
+}
+
+func (layer3NetConfInfo *Layer3NetConfInfo) GetMtu() int {
+	return layer3NetConfInfo.MTU
+}
+
+// GetNadName returns key of NetAttachDefInfo.NetAttachDefs map, also used as Pod annotation key
+func GetNadName(namespace, name string) string {
+	return fmt.Sprintf("%s/%s", namespace, name)
+}
+
+// GetSecondaryNetworkPrefix gets the string used as prefix of the logical entities
+// of the secondary network of the given network name, in the form of <netName>_.
+//
+// Note that for port_group and address_set, it does not allow the '-' character,
+// which will be replaced with ".". Also replace "/" in the nadName with "."
+func GetSecondaryNetworkPrefix(netName string) string {
+	name := strings.ReplaceAll(netName, "-", ".")
+	name = strings.ReplaceAll(name, "/", ".")
+	return name + "_"
+}
+
+func newNetConfInfo(netconf *ovncnitypes.NetConf) (NetConfInfo, error) {
+	var netconfInfo NetConfInfo
+
+	if !netconf.IsSecondary {
+		return nil, nil
+	}
+
+	if netconf.Topology == types.Layer3AttachDefTopoType {
+		netconfInfo = &Layer3NetConfInfo{
+			NetCidr: netconf.NetCidr,
+			MTU:     netconf.MTU,
+		}
+	} else {
+		// other topology nad can be supported later
+		return nil, fmt.Errorf("topology %s not supported", netconf.Topology)
+	}
+	err := netconfInfo.Verify()
+	if err != nil {
+		return nil, fmt.Errorf("%s netconf verification failed: %v", netconf.Topology, err)
+	}
+	return netconfInfo, nil
+}
+
+// ParseNADInfo parses config in NAD spec and return a NetAttachDefInfo object for secondary networks
+func ParseNADInfo(netattachdef *nettypes.NetworkAttachmentDefinition) (NetInfo, NetConfInfo, error) {
+	netconf, err := ParseNetConf(netattachdef)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	netconfInfo, err := newNetConfInfo(netconf)
+	if err != nil {
+		return nil, nil, err
+	}
+	return NewNetInfo(netconf), netconfInfo, nil
+}
+
+func NewNetInfo(netconf *ovncnitypes.NetConf) NetInfo {
+	// default network netInfo is nil
+	nInfo := (*NetNameInfo)(nil)
```

Comment:
why not just do "var nInfo *netNameInfo"? I think that is a more common practice. I've never seen the above done before.